src/pages/MainPage.vue

```diff
@@ -423,74 +365,45 @@ export default {
 
     this.updateFulfillment();
 
-    this.searchX = $("#searchInputTF").offset().left;
-    this.searchY = $("#searchInputTF").offset().top + $("#searchInputTF").outerHeight();
+    this.searchX = $('#searchInputTF').offset().left;
+    this.searchY = $('#searchInputTF').offset().top + $('#searchInputTF').outerHeight();
 
-    $(window).on("resize", function() {
-      this.searchX = $("#searchInputTF").offset().left;
-      this.searchY = $("#searchInputTF").offset().top + $("#searchInputTF").outerHeight();
+    $(window).on('resize', function () {
+      this.searchX = $('#searchInputTF').offset().left;
+      this.searchY = $('#searchInputTF').offset().top + $('#searchInputTF').outerHeight();
     }.bind(this));
 
     document.body.addEventListener('click', function (e) {
       this.showSearch = false;
     }.bind(this));
 
-    if(this.$cookies.isKey('dismissedOld')) {
+    if (this.$cookies.isKey('dismissedOld')) {
       this.dismissedOld = JSON.parse(this.$cookies.get('dismissedOld'));
-      this.cookiesAllowed = true;
+      this.$store.commit('allowCookies');
     }
-    if(this.$cookies.isKey('dismissedCookies')) {
+    if (this.$cookies.isKey('dismissedCookies')) {
       this.dismissedCookies = JSON.parse(this.$cookies.get('dismissedCookies'));
-      this.cookiesAllowed = true;
+      this.$store.commit('allowCookies');
     }
 
     // developer.mit.edu version commented out because I couldn't get it to work. filed an issue to resolve it.
     // axios.get('https://mit-course-catalog-v2.cloudhub.io/coursecatalog/v2/terms/2018FA/subjects', {headers:{client_id:'01fce9ed7f9d4d26939a68a4126add9b', client_secret:'D4ce51aA6A32421DA9AddF4188b93255'}})
     // , 'Accept': 'application/json'} ?
     // full=true is ~3x bigger but has some great info like "in_class_hours" and "rating"
-    axios.get(process.env.FIREROAD_URL + `/courses/all?full=true`)
-      .then(response => {
-        this.subjectsInfo = response.data;
-        this.genericCourses = this.makeGenericCourses();
-        this.subjectsIndexDict = this.subjectsInfo.reduce(function (obj, item, index) {
-          obj[item.subject_id] = index;
-          return obj;
-        }, {});
-        this.genericIndexDict = this.genericCourses.reduce(function (obj, item, index) {
-          obj[item.subject_id] = index;
-          return obj;
-        }, {});
-      });
+    this.$store.dispatch('loadAllSubjects').then(() => {
+      console.log('Subjects were loaded successfully!');
+    }).catch((e) => {
+      console.log('There was an error loading subjects: \n' + e);
+    });
```

Comment:
Is this for debugging or should we display this in console on prod?

Reply:
I think it would be nice to display errors that occur in the production console (provided we ensure we don't accidentally leak sensitive credentials). That way, we can just tell users to provide us with a log that might make debugging on our end easier.

Reply:
We talked about this the other day and decided that having just a few logs show in production would do more good than harm because it could make debugging way easier and if the site is already broken showing some error messages in the console isn't going to make things worse.